src/marqo/tensor_search/utils.py

```diff
@@ -137,3 +148,91 @@ def check_device_is_available(device: str) -> bool:
         return True
     else:
         return False
+
+
+
+
+
+def convert_to_MediaType(type_str: str) -> MediaType:
+    #This can actually be a dictionary
+    if type_str == "text":
+        return MediaType.text
+    elif type_str == "image":
+        return MediaType.image
+    elif type_str == "video":
+        return MediaType.video
+    else:
+        raise errors.MarqoError(f"Unspported type for {type_str}")
+
+def convert_to_FileType(type_str = "default") -> FileType:
+    #This can actually be a dictionary :(
+    if type_str == "default":
+        return FileType.default
+    elif type_str == "url":
+        return FileType.url
+    elif type_str == "www.youtube.com":
+        return FileType.youtube
+    elif type_str == "www.tiktok.com":
+        return FileType.tiktok
+    elif type_str == "local_path":
+        return FileType.local
+    elif type_str == "straight_text":
+        return FileType.straight_text
+    elif type_str == "ndarray":
+        return FileType.ndarray
+    elif type_str == "PILImage":
+        return FileType.PILImage
+    elif type_str == "ListOfPILImage":
+        return FileType.ListOfPILImage
+    else:
+        raise errors.MarqoError(f"Unsupported filetype for {type_str}")
+
+
+
+def get_filename_from_cd(cd):
+    """
+    Get filename from content-disposition
+    """
+    if not cd:
+        return None
+    fname = re.findall('filename=(.+)', cd)
+    if len(fname) == 0:
+        return None
+    return fname[0]
+
+def content_routering(field_content, infer_if_media = True):
+    if isinstance(field_content, str):
+        if infer_if_media:
+            if validators.url(field_content):
+                try:
+                    downloaded_file = wget.download(field_content)
```

Comment:
can we control the name of the downloaded file here?

Reply:
i think we might want to use temp dir for now

Reply:
A temp directory is used to store the downloaded files.

As for the filename, mimetypes uses the file name to determine the type. Maybe we should just keep the original name?